rtCommon/bidsArchive.py

```diff
@@ -618,6 +669,25 @@ def getIncremental(self, imageIndex: int = 0, **entities) \
                              "file; try specifying more to narrow to one match "
                              f"(expected 1, got {len(candidates)})")
 
+        # Confirmed that entities specify a unique run, so can use or fill cache
+        if useCache:
+            # Check if requested entities are subset of the ones in the cache
+            # If they aren't, then they identify a different run (cache miss)
+            #
+            # Subset check is imoprtant because incremental sometimes has extra
+            # entities (like datatype) that aren't necessarily needed to
+            # identify a single run, and thus a strict equality check fails.
+            if not (entities.items() <=
+                    self._getCache.getRunEntities().items()):
+                self._getCache = self.getBidsRun(**entities)
```

Comment:
What if the entities values passed in to getIncremental() are incomplete (underspecified). Would this subset match when it would be better to throw an error?

Reply:
The entity values passed to getIncremental(), at this stage of the method, are guaranteed to identify a unique run in the archive on disk (the code above throws an error if there is not exactly 1 match for the provided entities).

Being able to get an incremental without specifying all entities is intended behavior so that users don't (for example) have to specify the subject in an archive with only one subject, or the run when every subject only has one run for a given task.